### PR TITLE
Make use of verifyEntry result

### PR DESCRIPTION
**Pull request by crazybuster**
@haadcode  This is actually in reference to https://github.com/orbitdb/orbit-db-store/pull/28#issuecomment-405867980

Basically this is a small as possible commit so that you do not rely on exceptions for a VerifyEntry false. This would be nice so that if multiple heads are sent, you can still process the rest if one of the entries fail. I'm not sure how this effects: https://github.com/orbitdb/ipfs-log/pull/128

Would welcome any comments that you might have.

### CODE REVIEW

**Comment by haadcode** [comment]
@crazybuster thank you for the PR! Makes sense as per your comments.

Would it be possible to add a test for this to make sure that it works as expected?

**Comment by haadcode** [comment]
> I'm not sure how this effects: #128

It looks like the verification part will change with #128, but it would make sense to get this in regardless. I agree this is a better way to handle it and we can make sure before merging #128 that the logic is the same as you have proposed.

**Comment by crazybuster** [comment]
@haadcode  I added the test you requested. In adding the test I realized that it's actually a bigger security hole than I had thought. If you just put in any valid signature(one that passes the split into r and s) this will allow the entry to go through as valid since it was just looking for the exception before.

File: test/signed-log.spec.js [hunk 1]
```diff
@@ -262,5 +262,19 @@ apis.forEach((IPFS) => {
       assert.equal(log1.values.length, 1)
       assert.equal(log1.values[0].payload, 'one')
     })
+
+    it('throws an error if log is signed but the signature is valid and doesn\'t verify', async () => {
```
Comment (haadcode) [inline test/signed-log.spec.js]: Thank you for adding the test for this! ❤️ 

The description here says "throws an error..." but the assertions are not catching the error. I believe we need to:
```javascript
try {
// appends
// modify sig
// join
} catch (e) {
  err = e
}

assert.equal(err, `Error: Invalid signature in entry '${entry.hash}'`
```

Can you update the test to catch the error and verify it's thrown?

**Comment by crazybuster** [comment]
@haadcode Sorry, the test description was wrong, but yes that's exactly the case! When it verifies a valid, but wrong(ie signed by a different key) signature. It will return false instead of throwing an error. This will allow wrong signatures to pass. The previous test just creates an invalid signature when it mixes up the bits. But it's possible to create a valid but still wrong signature. Will update the test description now.

**Comment by crazybuster** [comment]
@haadcode is there anyway to get this merged in quicker. There's a few cases where I'd actually like this in our current implementation.

**Comment by haadcode** [comment]
Thanks for fixing the test description @crazybuster! LGTM to merge.

**Comment by haadcode** [comment]
Thank you for the PR @crazybuster, much appreciated! Will make a new release to npm soon.

**Comment by haadcode** [comment]
Published to [npm as 4.2.0](https://www.npmjs.com/package/ipfs-log).

**Pull request merged**